src/app/dataset-view/additional-components/dataset-settings-component/tabs/general/dataset-settings-general-tab.component.ts

```diff
@@ -103,22 +109,58 @@ export class DatasetSettingsGeneralTabComponent extends BaseComponent implements
     public deleteDataset(): void {
         const datasetId = this.datasetBasics.id;
         const accountId = this.datasetBasics.owner.id;
-        promiseWithCatch(
-            this.modalService.error({
-                title: "Delete",
-                message: "Do you want to delete a dataset?",
-                yesButtonText: "Ok",
-                noButtonText: "Cancel",
-                handler: (ok) => {
-                    if (ok) {
-                        this.datasetSettingsService
-                            .deleteDataset(accountId, datasetId)
-                            .pipe(takeUntilDestroyed(this.destroyRef))
-                            .subscribe();
+
+        this.datasetApi.datasetPushSyncStatuses(datasetId).subscribe((data: DatasetPushSyncStatusesQuery) => {
```

Comment:
We can't use API directly in the component. We need to add a method in the service

Reply:
done